File: src/runtime/cs/std/math_native.cs
using System;

namespace Pytra.CsModule
{
    // Generated std/math.cs delegates host bindings through this native seam.
    public static class math_native
    {
        public static double pi { get { return Math.PI; } }
        public static double e { get { return Math.E; } }

        public static double sqrt(double x)
        {
            return Math.Sqrt(x);
        }

        public static double sin(double x)
        {
            return Math.Sin(x);
        }

        public static double cos(double x)
        {
            return Math.Cos(x);
        }

        public static double tan(double x)
        {
            return Math.Tan(x);
        }

        public static double exp(double x)
        {
            return Math.Exp(x);
        }

        public static double log(double x)
        {
            return Math.Log(x);
        }

        public static double log10(double x)
        {
            return Math.Log10(x);
        }

        public static double fabs(double x)
        {
            return Math.Abs(x);
        }

        public static double floor(double x)
        {
            return Math.Floor(x);
        }

        public static double ceil(double x)
        {
            return Math.Ceiling(x);
        }

        public static double pow(double x, double y)
        {
            return Math.Pow(x, y);
        }
    }
}

File: src/runtime/julia/std/math_native.jl
# Native implementation for pytra.std.math
# Generated std/math.jl delegates host bindings through this native seam.

module __MathNative
    const pi = Float64(Base.MathConstants.pi)
    const e = Float64(Base.MathConstants.e)

    function sqrt(x)
        return Base.sqrt(x)
    end
    function sin(x)
        return Base.sin(x)
    end
    function cos(x)
        return Base.cos(x)
    end
    function tan(x)
        return Base.tan(x)
    end
    function exp(x)
        return Base.exp(x)
    end
    function log(x)
        return Base.log(x)
    end
    function log10(x)
        return Base.log10(x)
    end
    function fabs(x)
        return Base.abs(x)
    end
    function floor(x)
        return Int(Base.floor(x))
    end
    function ceil(x)
        return Int(Base.ceil(x))
    end
    function pow(b, e)
        return b ^ e
    end
end
